cogito/protocol.go

```diff
@@ -212,18 +217,47 @@ func (src *Source) UnmarshalJSON(data []byte) error {
 // Validate verifies the Source configuration and applies defaults.
 func (src *Source) Validate() error {
 	//
-	// Validate mandatory fields.
+	// Evaluate mandatory fields.
 	//
 	var mandatory []string
-	if src.Owner == "" {
-		mandatory = append(mandatory, "owner")
+	githubRequired := map[string]string{
+		"Owner":       "owner",
+		"Repo":        "repo",
+		"AccessToken": "access_token",
+	}
+
+	defaultSinks := []string{"gchat", "github"}
+	defaultSinksSet := sets.From(defaultSinks...)
+	sinksSet := sets.From(src.Sinks...)
+
+	if sinksSet.Size() > 0 {
+		sinksNotValid := sinksSet.Difference(defaultSinksSet)
+		if sinksNotValid.Size() > 0 {
+			return fmt.Errorf("source: invalid sink: %s. Supported sinks: %s", sinksNotValid, defaultSinks)
+		}
 	}
-	if src.Repo == "" {
-		mandatory = append(mandatory, "repo")
+
+	v := reflect.ValueOf(src).Elem()
```

Comment:
I think @marco-m-pix4d will not like this :)

Reply:
May be, I tried to address the suggestion from @odormond , may be I could obtain the same differently.